### ISSUE TITLE
support different #lang reader languages

### ISSUE THREAD

**Issue by stchang** [stchang/pasterack#34]
Right now, something like `#lang at-exp racket` does not work. Thanks @greghendershott.

**Comment by stchang** [comment]
Suggestion from @takikawa: http://pasterack.org/pastes/66105

```
Welcome to Racket v6.1.1.6.
-> (parameterize ([read-accept-lang #t] [read-accept-reader #t]) (read))
#; ()
#lang at-exp racket
@+[1 2]
'(module anonymous-module racket (#%module-begin (+ 1 2)))
```

**Comment by stchang** [comment]
The suggestion works for the eval part.

For the typesetting part, can throw everything from the paste (including `#lang`, with custom readers, and leading comments) into `codeblock`. But the issue is getting the `for-label` links, which requires manually parsing both the `#lang` and the additional requires.

01/09Fri17:17 < stchang> asumu: thanks that works for the eval part, but i 
                         think i still have to do some manual #lang parsing in 
                         order to get scribble to typeset and link it properly
01/09Fri17:18 < asumu> stchang: oh blargh, right
01/09Fri17:18 < stchang> specifically, i think i need to manually parse the 
                         #lang and the requires
01/09Fri17:18 < asumu> But then can you use codeblock or whatever it's called 
                       to handle non-sexp readers?
01/09Fri17:18 < stchang> yeah there's "code"
01/09Fri17:19 < stchang> which accepts a #:lang argument
01/09Fri17:19 < stchang> but i would still have to parse out the string to pass 
                         as that argument
01/09Fri17:19 < stchang> and i have to extract requires, so i can do a 
                         for-label require
01/09Fri17:21 < stchang> asumu: hmm, actually, you're right about codeblock
01/09Fri17:21 < stchang> i can just give everything, including the #lang to 
                         codeblock
01/09Fri17:21 < stchang> but that doesnt handle the for-label requires
01/09Fri17:22 < asumu> Oh hum, right it doesn't link
01/09Fri17:22 < asumu> Can you just use your current rendering if you can 
                       detect #lang racket (or similar) but use codeblock 
                       otherwise?
01/09Fri17:23 < stchang> yes maybe that would work
01/09Fri17:23 < asumu> Oh are you saying the for-label requires still needs you 
                       to parse the #lang?
01/09Fri17:23 < stchang> yes, right now i parse the #lang and include it in a 
                         for-label require
01/09Fri17:23 < stchang> as well as the other requires
01/09Fri17:24 < stchang> but you're right, there's not reason why i cant just 
                         keep what im already doing
01/09Fri17:24 < stchang> except for the leading comment issue

**Comment by stchang** [comment]
fixed with 52c3f18e57ed29d072a35ce8cad68f40fef6e973
